quint/src/runtime/compile.ts

```diff
@@ -60,6 +60,8 @@ export interface CompilationContext {
 export interface CompilationState {
   // The ID generator used during compilation.
   idGen: IdGenerator
+  // A list of modules as they are constructed, without flattening.
```

Comment:
Sorry to be a stickler, but I think it would help for this comment to note why this is needed. E.g., something like

```suggestion
  // A list of modules as they are constructed, without flattening. This is needed to derive correct name resolution during incremental compilation in a flattened context. 
```

My thought is that this info will then make sense of all the later uses. We of course risk the comment growing stale, but we always do. :)

the principle motivating my suggestion is that we want the comments to tell help us understand the why, and we mostly leave it to the code to self-document the what.

Reply:
Makes a lot of sense! Thanks